src/OpenTelemetry.Exporter.Zipkin/ZipkinActivityExporter.cs

```diff
@@ -0,0 +1,220 @@
+﻿// <copyright file="ZipkinActivityExporter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Sockets;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Exporter.Zipkin.Implementation;
+using OpenTelemetry.Trace.Export;
+
+namespace OpenTelemetry.Exporter.Zipkin
+{
+    /// <summary>
+    /// Zipkin exporter.
+    /// </summary>
+    public class ZipkinActivityExporter : ActivityExporter
+    {
+        private readonly ZipkinTraceExporterOptions options;
+        private readonly HttpClient httpClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ZipkinActivityExporter"/> class.
+        /// </summary>
+        /// <param name="options">Configuration options.</param>
+        /// <param name="client">Http client to use to upload telemetry.</param>
+        public ZipkinActivityExporter(ZipkinTraceExporterOptions options, HttpClient client = null)
+        {
+            this.options = options;
+            this.LocalEndpoint = this.GetLocalZipkinEndpoint();
+            this.httpClient = client ?? new HttpClient();
+        }
+
+        internal ZipkinEndpoint LocalEndpoint { get; }
+
+        /// <inheritdoc/>
+        public override async Task<ExportResult> ExportAsync(IEnumerable<Activity> batchActivity, CancellationToken cancellationToken)
+        {
+            try
+            {
+                await this.SendBatchActivityAsync(batchActivity).ConfigureAwait(false);
```

Comment:
This part needs to be revisited since it doesn't seem to be deterministic. Not a blocker for this PR though.